src/main/java/com/codessquad/qna/controller/PostController.java

```diff
@@ -49,33 +57,61 @@ public String getAllPosts(Model model) {
     /**
      * 매개변수로 오는 id 값을 기반으로 해당 포스트를 불러옵니다.
      * 만약 해당 유저가 없다면 CanNotFindPostException를 리턴합니다.
-     * @param id
+     * @param id Post Id
      * @param model
      * @throws CanNotFindPostException
      * @return
      */
-    @GetMapping("/questions/{id}")
+    @GetMapping("/{id}")
     public String getPost(@PathVariable Long id, Model model) {
-        try{
-            Post post = postService.getPost(id);
-            model.addAttribute("post", post);
-        }catch (CanNotFindPostException e){
-            logger.error(e.getMessage());
-        }
+        model.addAttribute("post", postService.getPost(id));
         return "qna/show";
     }
 
     /**
-     * IllegalArgumentException 을 핸들링 해주는 메소드
-     * 지금은 그냥 redirect 를 하지만 나중에는 다른 페이지로 넘겨주는 작업이 필요할듯함.
-     * @param e
+     * Post UpdateForm 으로 이동할 수 있음.
+     * 세션에 로그인 되어 있는 유저와 작성자를 비교하여 틀릴시 IllegalAccessException 을 리턴함
+     * @param id Post id
+     * @param model
+     * @throws IllegalAccessException
+     * @return
+     */
+    @GetMapping("/{id}/form")
+    public String updatePostForm(@PathVariable Long id, HttpSession httpSession, Model model) throws IllegalAccessException {
+        Post post = postService.getPost(id);
+        if(!post.isMatchedAuthor(HttpSessionUtils.getUserFromSession(httpSession))){
+            throw new IllegalAccessException("다른 사람의 글을 수정할 수 없습니다");
+        }
+        model.addAttribute("post", post);
+        return "qna/updateForm";
+    }
+
+    /**
+     * postDto 를 받아 기존의 post 를 업데이트 할 수 있도록 하였음
+     * @param id Post id
+     * @param postDto
+     * @return
+     */
+    @PutMapping("/{id}")
+    public String updatePost(@PathVariable Long id, @ModelAttribute PostDto postDto) {
+        postService.updatePost(id, postDto);
+        return "redirect:/";
+    }
+
+    /**
+     * 해당 id 에 Mapping 되어 있는 게시물을 삭제합니다.
+     * 작성자가 아니라면 IllegalAccessException 이 발생합니다.
+     * @param id Post id
      * @return
      */
-    @ExceptionHandler(IllegalArgumentException.class)
-    public String handleException(Exception e, Model model) {
-        logger.error(e.getMessage());
-        model.addAttribute("exception", e);
-        return "error";
+    @DeleteMapping("/{id}")
+    public String deletePost(@PathVariable Long id, HttpSession httpSession) throws IllegalAccessException {
+        Post post = postService.getPost(id);
+        if(!post.isMatchedAuthor(HttpSessionUtils.getUserFromSession(httpSession))){
```

Comment:
```suggestion
        User sessionUser = HttpSessionUtils.getUserFromSession(httpSession)
        if(!post.isMatchedAuthor(sessionUser)) {
```
이렇게 써주는 편이 더 읽기 좋지 않나요?